### ISSUE TITLE
CPS Tax-Unit File: Potential Wage distribution discrepancy between SAS and Python Scripts

### ISSUE THREAD

**Issue by Amy-Xu** [open-source-economics/taxdata#144]
I'm inspecting the [updated UBI results](http://nbviewer.jupyter.org/github/andersonfrailey/Notebook-Uploads/blob/master/CPS%20UBI.ipynb) that Anderson created from CPS tax unit file, in order to update the [working paper](https://github.com/open-source-economics/UBI-examples/blob/master/ex1/UBI-working-paper.pdf). In the working paper, we used tax liabilities from PUF and combined it with CPS benefits at income group levels. In the updated version, we switched to CPS tax unit file for the tax liabilities. Thus some degree of difference should be expected because PUF and CPS tax unit files are simply not the same.

However I find it troublesome that the wage distribution for under 65 population shifted upward quite a bit. [Table 4](https://github.com/open-source-economics/UBI-examples/blob/master/ex1/UBI-working-paper.pdf) in the working paper indicates 47 million units with wage between 0 and 10k, while [the updated table (3rd)](http://nbviewer.jupyter.org/github/andersonfrailey/Notebook-Uploads/blob/master/CPS%20UBI.ipynb) shows only 35 million units in the same bin. Meanwhile, the total unit numbers from these two versions are very close (129 vs 128 million), the discrepancy in the lowest income bin means the wage distribution gets pushed toward the upper end. 

Although this might seem to be a difference between PUF and CPS at first sight, in essence it's about two versions of CPS tax unit data. In the working paper, the [old CPS file](https://github.com/open-source-economics/UBI-examples/blob/master/ex1/notebook/cps_benefit.csv) renders about 49 million units for that bin (wage 0-10k, under 65), which is consistent with PUF results. But the updated CPS files only has 35 million. This discrepancy is quite large in my opinion given that it is only one income bin for under 65 population. 

Last time when I discussed this issue with Anderson, I suspected the discrepancy resulted from 1 year CPS input (for the working paper) vs three year CPS input (for the current tax-unit file). But I ruled out this possibility by running the Python scripts for tax unit creation in PR #118. Using only 2014 CPS vs three year (2013-2015) CPS yields consistent number of tax units in that bin (30.7 million vs 30.8 million). 

So I scratched my head for a few days and thought maybe there's a glitch in the translation, because the CPS file used for the working paper was generated through SAS scripts while the new version is from Python. 

Updating the working paper is a part of the grant requirement due by March. I would love to hear thoughts or suggestions how to proceed with this distribution issue.

@andersonfrailey @MattHJensen @martinholmer @hdoupe

**Comment by Amy-Xu** [comment]
A quick update on this issue. I checked two earlier versions of CPS that produced by John (SAS) -- both files are based on 2015 CPS and yield similar number of units (33 - 35 million) for the 0-to-10k-wage-under-65 bin. I also checked a later version of three-year CPS, also produced by John (SAS), and the result is also the same as the other two. 

I think this is a good news -- it means there's little discrepancy between our translation and John's original SAS scripts. This can also be a bad news in two ways: 1) CPS tax units file might inherently have smaller number low-income units; 2) the version of CPS file we used for the working paper, which is the version most comparable to PUF in terms of this income bin, is not consistent with any other CPS files we have.

I'm still combing through the SAS scripts to see whether I could dig something out. Let me know if you see loopholes in my thoughts above.

@andersonfrailey @MattHJensen @martinholmer @hdoupe
